src/fsharp/TypeChecker.fs

```diff
@@ -14357,7 +14357,8 @@ module EstablishTypeDefinitionCores =
         let isStructRecordOrUnionType = 
             match synTyconRepr with
             | SynTypeDefnSimpleRepr.Record _ 
-            | SynTypeDefnSimpleRepr.Union _ -> 
+            | SynTypeDefnSimpleRepr.Union _
+            | SynTypeDefnSimpleRepr.TypeAbbrev _ -> 
```

Comment:
We should match `TypeAbbriv` because simple unions like `type DU = Case` are represented so in AST:

```
TypeDefn
                 (ComponentInfo
                    ([{TypeName = LongIdentWithDots ([Struct],[]);
                       ArgExpr =
                        Const
                          (Unit,
                           C:\Users\vaski\onedrive\документы\visual studio 2017\Projects\ConsoleApplication3\ConsoleApplication3\File2.fs (3,2--3,8) IsSynthetic=false);
                       Target = None;
                       AppliesToGetterAndSetter = false;
                       Range =
                        C:\Users\vaski\onedrive\документы\visual studio 2017\Projects\ConsoleApplication3\ConsoleApplication3\File2.fs (3,2--3,8) IsSynthetic=false;}],
                     [],[],[DU],
                     PreXmlDoc
                       (Microsoft.FSharp.Compiler.Range+pos,
                        Microsoft.FSharp.Compiler.Ast+XmlDocCollector),false,
                     None,
                     C:\Users\vaski\onedrive\документы\visual studio 2017\Projects\ConsoleApplication3\ConsoleApplication3\File2.fs (4,5--4,7) IsSynthetic=false),
                  Simple
                    (TypeAbbrev
                       (Ok,LongIdent (LongIdentWithDots ([Case],[])),
                        C:\Users\vaski\onedrive...
```